src/index.ts

```diff
@@ -0,0 +1,125 @@
+import { SecretManagerServiceClient, protos } from "@google-cloud/secret-manager";
+
+/**
+ * Filter function to filter secrets by name that dont have to be loaded
+ */
+type FilterFunction = (name: string) => boolean;
+
+/**
+ * Definition of possible options for the module
+ */
+type WithGoogleSecretsOptions = {
+  /**
+   * The project name in google could to load the secrets from
+   */
+  projectName: string;
+
+  /**
+   * A mapping to define which secret should overwritte which config entry (secretName, configEntry(ies))
+   */
+  mapping: Record<string, string | string[]>;
+
+  /**
+   * The filter to either filter directly by labels (string) or by a function
+   */
+  filter?: string | FilterFunction;
```

Comment:
Can we split this up into `filter` and `filterFn`? 
There are use cases where you want to have both (server side filter and client side filtering).

Reply:
of course, added two separate parameters👍